closure-deps/lib/depgraph.js

```diff
@@ -231,7 +231,8 @@ class Graph {
       if (this.depsByPath.has(dep.path)) {
         throw new Error('File registered twice? ' + dep.path);
       }
-      this.depsByPath.set(dep.path, dep);
+      this.depsByPath.set(
+          path.join(path.dirname(dep.path), path.basename(dep.path)), dep);
```

Comment:
This doesn't make the key an absolute path. This code should just return end up being `dep.path` again. `path.dirname('./a/b')` will be `./a`, fyi.

`path.resolve` will make a path absolute. `path.join` won't.

`path.join('../a', 'b')` will be `../a/b`. `path.resolve('../a', 'b')` will be some absolute path ending with `/a/b`.

https://nodejs.org/docs/latest/api/path.html#path_path_resolve_paths
https://nodejs.org/docs/latest/api/path.html#path_path_join_paths


tl;dr I think that when a `Dependency` is created you should say `this.path = path.resolve(filePath)` in the constructor. I think that should force all `Dependency`s to have absolute paths, meaning this map will use absolute paths, and `PathModuleResolver` can remain unchanged.